### PR TITLE
patch_manager: Add support for disabling patches

### PR DESCRIPTION
**Pull request by DarkLordZach**
This is a temporary workaround to #1816, as well as any other game that fails to work with its update. The workaround is to, with this PR, open the properties and disable the update.

This adds the following properties dialog, which can be displayed from the game list via right-clicking:
image

This was recycled from #1518, as that PR had other issues, but this is one of the good aspects of it. The UI will display some statistics about the game, its icon, and a list of add-ons and their versions. Unchecking the box will disable the add-on. When disabled, a `[D]` will display in front of the addon in the game list.

### CODE REVIEW

File: src/yuzu/configuration/configure_per_general.cpp [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+#include "core/file_sys/patch_manager.h"
+#include "core/file_sys/xts_archive.h"
+#include "core/loader/loader.h"
+#include "input_common/main.h"
+#include "yuzu/configuration/config.h"
+#include "yuzu/configuration/configure_input.h"
+#include "yuzu/configuration/configure_per_general.h"
+#include "yuzu/ui_settings.h"
+
+ConfigurePerGameGeneral::ConfigurePerGameGeneral(u64 title_id, QWidget* parent)
```
Comment (lioncash) [inline src/yuzu/configuration/configure_per_general.cpp]: parent should be first in the parameter list.

File: src/yuzu/configuration/configure_per_general.cpp [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    layout->setSpacing(0);
+    layout->addWidget(tree_view);
+
+    ui->scrollArea->setLayout(layout);
+
+    scene = new QGraphicsScene;
+    ui->icon_view->setScene(scene);
+
+    connect(item_model, &QStandardItemModel::itemChanged,
+            [&]() { UISettings::values.is_game_list_reload_pending.exchange(true); });
```
Comment (lioncash) [inline src/yuzu/configuration/configure_per_general.cpp]: `()` and `&` aren't necessary.

File: src/yuzu/configuration/configure_per_general.cpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    } else {
+        std::string title;
+        if (loader->ReadTitle(title) == Loader::ResultStatus::Success)
+            ui->display_name->setText(QString::fromStdString(title));
+
+        std::string developer;
+        if (loader->ReadDeveloper(developer) == Loader::ResultStatus::Success)
+            ui->display_developer->setText(QString::fromStdString(developer));
+
+        ui->display_version->setText("1.0.0");
```
Comment (lioncash) [inline src/yuzu/configuration/configure_per_general.cpp]: `QStringLiteral("1.0.0")`

File: src/yuzu/configuration/configure_per_general.h [hunk 4]
```diff
@@ -0,0 +1,56 @@
+﻿// Copyright 2016 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <functional>
```
Comment (lioncash) [inline src/yuzu/configuration/configure_per_general.h]: Quite a lot of these includes aren't necessary and it's indicative of copy/pasting another file. Can you... not do that sort of thing in the future?

Comment (DarkLordZach) [reply src/yuzu/configuration/configure_per_general.h]: understood

**Pull request merged**